frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -586,3 +594,23 @@ func contains(s []uint32, e uint32) bool {
 	}
 	return false
 }
+
+func isRunningOnGCE() bool {
+	_, err := net.LookupIP("metadata.google.internal")
+	return err == nil
+}
+
+// For instances running on GCE, checks whether the instance was created with a service account having an access token.
+func hasServiceAccountAccessToken() (bool, error) {
+	req, err := http.NewRequest("GET", "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/token", nil)
+	if err != nil {
+		return false, err
+	}
+	req.Header.Set("Metadata-Flavor", "Google")
```

Comment:
Is this required? I was under the impression that this is something the server sets in its responses, not something you had to set in the request.

Reply:
yes, https://cloud.google.com/compute/docs/metadata/querying-metadata#parts-of-a-request